static/src/javascripts/projects/common/modules/experiments/ab-ophan.js

```diff
@@ -0,0 +1,151 @@
+
+const submit = (payload: OphanABPayload): void =>
+    ophan.record({
+        abTestRegister: payload,
+    });
+
+/**
+ * generate an A/B event for ophan
+ */
+const makeABEvent = (
+    variant: Variant,
+    complete: string | boolean
+): OphanABEvent => {
+    const event: OphanABEvent = {
+        variantName: variant.id,
+        complete,
+    };
+
+    if (variant.options && variant.options.campaignCodes) {
+        event.campaignCodes = variant.options.campaignCodes;
+    }
+
+    return event;
+};
+
+/**
+ * Checks if this test will defer its impression by providing its own impression function.
+ *
+ * If it does, the test won't be included in the Ophan call that happens at pageload, and must fire the impression
+ * itself via the callback passed to the `impression` property in the variant.
+ *
+ * @param test
+ * @returns {boolean}
+ */
+const defersImpression = (test: ABTest): boolean =>
+    test.variants.every(variant => typeof variant.impression === 'function');
+
+/**
+ * Create a function that will fire an A/B test to Ophan
+ *
+ * @param test
+ * @param {string} variantId
+ * @param {boolean} complete
+ * @returns {Function} to fire the event
+ */
+const buildOphanSubmitter = (
+    test: ABTest,
+    variantId: string,
+    complete: boolean
+): (() => void) => {
+    const data = {};
+    const variant = abUtils.getVariant(test, variantId);
+
+    if (variant) data[test.id] = makeABEvent(variant, String(complete));
+
+    return () => submit(data);
+};
+
+/**
+ * Create a function that sets up listener to fire an Ophan `complete` event. This is used in the `success` and
+ * `impression` properties of test variants to allow test authors to control when these events are sent out.
+ *
+ * @see {@link defersImpression}
+ * @param {Boolean} complete
+ * @returns {Function}
+ */
+const registerCompleteEvent = complete => test => {
+    const variantId = abUtils.getTestVariantId(test.id);
+
+    if (variantId && variantId !== 'notintest') {
+        const variant = abUtils.getVariant(test, variantId);
+
+        if (variant != null) {
+            const listener =
+                (complete ? variant.success : variant.impression) || noop;
+
+            try {
+                listener(buildOphanSubmitter(test, variantId, complete));
+            } catch (err) {
+                reportError(err, {}, false);
+            }
+        }
+    }
+};
+
+export { buildOphanSubmitter };
+
+export const registerCompleteEvents = (tests: Array<ABTest>): void =>
+    tests.forEach(registerCompleteEvent(true));
+
+export const registerImpressionEvents = (tests: Array<ABTest>): void =>
+    tests.filter(defersImpression).forEach(registerCompleteEvent(false));
+
+export const buildOphanPayload = (): OphanABPayload => {
+    try {
+        const log = {};
+        const serverSideTests = Object.keys(config.tests).filter(
+            test => !!config.tests[test]
+        );
+
+        getActiveTests()
+            .filter(not(defersImpression))
```

Comment:
Yes I guess, otherwise you can still do that I think:
```
const and = (...args: any[]): boolean => (f, g) => f(...args) && g(...args)
```
but that may be more difficult to read